### PR TITLE
Porting PIPE from AliRoot

### PR DESCRIPTION
**Pull request by sawenzel**
providing the beam pipe based on AliPIPEupgrade
from AliRoot ad01f46ea3edabdc11

### CODE REVIEW

File: Detectors/Passive/src/Pipe.cxx [hunk 1]
```diff
@@ -8,99 +8,2921 @@
...
+#include <TGeoPcon.h>
+#include <TGeoTorus.h>
+#include <TGeoTube.h>
+#include <TVirtualMC.h>
+#include "TGeoManager.h"  // for TGeoManager, gGeoManager
+#include "TGeoMaterial.h" // for TGeoMaterial
+#include "TGeoMedium.h"   // for TGeoMedium
+#include "TGeoVolume.h"   // for TGeoVolume
+#ifdef NDEBUG
+#undef NDEBUG
```
Comment (matthiasrichter) [inline Detectors/Passive/src/Pipe.cxx]: Why is this necessary?

File: Detectors/Passive/src/Pipe.cxx [hunk 2]
```diff
@@ -8,99 +8,2921 @@
...
+  //////////////////// NEW BEAM PIPE GEOMETRY FOR MuonForwardTracker ,
+  // Authors: F. Manso, R. Tieulent
+  // Drawings from C. Gargiulo :
+  // \\cern.ch\dfs\Workspaces\c\cgargiul\EXPERIMENT\ALICE\ALICE_MECHANICS\ALICE_DATA_PACKAGE\IN\DETECTORS\ITS_UPGRADE\1-DESIGN\3D_cad_model\R14_20140311_ALI\
+  //
+  //------------------- Pipe version 4.7 March 2014 -----------------------------
+
+  TGeoVolumeAssembly* beamPipeCsideSection = new TGeoVolumeAssembly("BeamPipeCsideSection");
+
+  Float_t fBeryliumSectionOuterRadius = 1.9;
```
Comment (matthiasrichter) [inline Detectors/Passive/src/Pipe.cxx]: the `f` in front probably indicates a float, but since some conventions use this for members of classes this might be misleading. It looks like that several conventions are used in this class, further down one finds similar definitions with `k` prefix, and also some without prefix. Last but not least: should many of those variables be `const` or even `constexpr`?

File: Detectors/Passive/src/Pipe.cxx [hunk 3]
```diff
@@ -8,99 +8,2921 @@
...
+  Float_t fFlangeRmax = 4.3;
+  Float_t fFlangeLength = 1.4;
+
+  Float_t fBellowPlieRadius = 0.17;    // radius of bellow plies
+  Float_t fBellowPlieThickness = 0.03; // Thickness of bellow plies 300 microns
+  Int_t fNBellowConvolutions = 7;
+
+  // If user set Rmax=0/Thick=0 use defaults, else use user input
+  if (mBePipeRmax > 0.)
+    fBeryliumSectionOuterRadius = mBePipeRmax;
```
Comment (matthiasrichter) [inline Detectors/Passive/src/Pipe.cxx]: please check if there is a potential problem here: `fDzFirstCone` is calculated from the default and used without adjustment after setting the user define outer radius

File: Detectors/Passive/src/Pipe.cxx (hunk 1 above)
Comment (sawenzel) [reply Detectors/Passive/src/Pipe.cxx]: To force assert being available in any case.

File: Detectors/Passive/src/Pipe.cxx (hunk 2 above)
Comment (sawenzel) [reply Detectors/Passive/src/Pipe.cxx]: I know. This is indeed highly misleading. For the moment this is a straight copy-paste from AliRoot. Fixing all the variable names will take some time (I need a tool for that).

**Comment by sawenzel** [comment]
@dberzano : The log for the macos failure does not seem to be shown.

**Comment by dberzano** [comment]
@sawenzel: as usual, this was a temporary error and it will clear out. I have deleted the empty message as I was restarting the checker due to the error you've reported elsewhere. Just try to be zen in the meanwhile ;-)

**Comment by sawenzel** [comment]
@matthiasrichter : I provided a revised version addressing your comments.

**Comment by matthiasrichter** [comment]
@sawenzel thanks, looks good
